bootstrap/src/io/appium/android/bootstrap/handler/ConfiguratorHandler.java

```diff
@@ -0,0 +1,64 @@
+import io.appium.android.bootstrap.AndroidCommand;
+import io.appium.android.bootstrap.AndroidCommandResult;
+import io.appium.android.bootstrap.CommandHandler;
+import org.json.JSONException;
+
+import java.util.Hashtable;
+
+import static io.appium.android.bootstrap.utils.API.API_18;
+
+/**
+ * This handler set {@link com.android.uiautomator.core.Configurator} related settings including
+ * ActionAcknowledgmentTimeout, KeyInjectionDelay, ScrollAcknowledgmentTimeout, WaitForIdleTimeout and WaitForSelectorTimeout
+ * <br/>command example {"cmd" :"action", "params":{"config":"setActionAcknowledgmentTimeout","value":5000}}
+ * <br/>example is the same as invoking {@link com.android.uiautomator.core.Configurator#setActionAcknowledgmentTimeout} to 5 seconds
+ */
+public class ConfiguratorHandler extends CommandHandler {
+    @Override
+    public AndroidCommandResult execute(AndroidCommand command) throws JSONException {
+        if (API_18) {
+            final Hashtable<String, Object> params = command.params();
+            int value = 0;
+            if (params.containsKey("value")) {
+                value = (Integer) params.get("value");
+            }
+            String methodName = ((String) params.get("method"));
+            //TODO: use reflection to invoke method would be more expandable; but Configurator is singleton
+            Configurator configurator = Configurator.getInstance();
+            switch (methodName) {
+                case "setActionAcknowledgmentTimeout":
+                    if (value == 0) // set to default when no value or zero
+                        value = 3000;
+                    configurator.setActionAcknowledgmentTimeout(value);
+                    break;
+                case "setKeyInjectionDelay":
+                    //KeyInjectionDelay default is 0
+                    configurator.setKeyInjectionDelay(value);
+                    break;
+                case "setScrollAcknowledgmentTimeout":
+                    if (value == 0) // set to default when no value
+                        value = 200;
+                    configurator.setScrollAcknowledgmentTimeout(value);
+                    break;
+                case "setWaitForIdleTimeout":
+                    if (value == 0) // set to default when no value
+                        value = 10000;
+                    configurator.setWaitForIdleTimeout(value);
+                    break;
+                case "setWaitForSelectorTimeout":
+                    if (value == 0) // set to default when no value
+                        value = 10000;
+                    configurator.setWaitForSelectorTimeout(value);
+                    break;
+                default:
+                    return getErrorResult("'configurator' command must contain 'type' key");
+            }
+            return getSuccessResult(value);
+        } else {
+            return getErrorResult("Device does not support API >= 18!");
```

Comment:
Yep, I copied this line from somewhere in this repo. did not check for the message content 😅